app/main/notify.py

```diff
@@ -1,48 +1,102 @@
-    }
+    personalisation = kwargs
+    if file:
+        personalisation["link_to_file"] = prepare_upload(file)
     try:
         notify_client = NotificationsAPIClient(Config.NOTIFY_API_KEY)
         notify_client.send_email_notification(
             email_address=email_address,
-            template_id=Config.CONFIRMATION_EMAIL_TEMPLATE_ID,
+            template_id=template_id,
             personalisation=personalisation,
         )
-        logging.info("Confirmation email sent via GovUK Notify")
+        current_app.logger.info("Email sent via GovUK Notify")
     except HTTPError as notify_exc:
-        logging.error(
-            f"HTTPError when trying to send confirmation email: params={personalisation} exception={notify_exc}"
+        current_app.logger.error(
+            f"HTTPError when trying to send email: params={personalisation} exception={notify_exc}"
+        )
+
+
+def prepare_upload(file: FileStorage):
+    """Wraps notify.prepare_upload in some exception handling / logging.
+
+    The file size limit is 2MB - files larger than this will not be sent.
+
+    :param file: file to prepare
+    :return: None
+    """
+    try:
+        return notify.prepare_upload(file)
+    except ValueError as err:
+        current_app.logger.error(f"Submitted file is too large to send via email - {err}")
+
+
+def send_confirmation_emails(excel_file: FileStorage, metadata: dict, user_email: str):
+    """Sends a confirmation email to the LA that submitted and to TF (with a link to the submission).
+
+    :param excel_file: Excel file that has been submitted successfully
+    :param metadata: contains information about the submission
+    :param user_email: the user's email address to send confirmation to
+    :return: None
+    """
+    personalisation = get_personalisation(excel_file, metadata)
+    current_app.logger.info("Sending confirmation emails to LA and TF")
+    # to the Local Authority
+    send_email(email_address=user_email, template_id=Config.LA_CONFIRMATION_EMAIL_TEMPLATE_ID, **personalisation)
+    # to Towns Fund - includes the file
+    send_email(
+        email_address=Config.TF_CONFIRMATION_EMAIL_ADDRESS,
+        template_id=Config.TF_CONFIRMATION_EMAIL_TEMPLATE_ID,
+        file=excel_file,
+        **personalisation,
+    )
+
+
+def get_personalisation(excel_file: FileStorage, metadata: dict):
+    """Builds email personalisation from the file, metadata and application config.
+
+    If the metadata is missing some values used to personalise then an error is logged.
+
+    :param excel_file: a file with a filename
+    :param metadata: metadata to retrieve personalisation details from
+    :return: the personalisation dictionary
+    """
+    personalisation = {
+        "name_of_fund": Config.FUND_NAME,
+        "reporting_period": Config.REPORTING_PERIOD,
+        "filename": excel_file.filename,
+        "place_name": metadata.get("Programme Name", ""),
+        "fund_type": metadata.get("FundType_ID", ""),
```

Comment:
Do we not want to pass a more friendly representation of the fund name to the email?

Reply:
yep good idea